js/private/js_run_devserver.mjs

```diff
@@ -54,6 +54,28 @@ function isNodeModulePath(srcPath) {
     return false;
 }
 
+// Returns the correct symlink type if we are building on windows. This doesn't appear to be an issue
+// on *nix based machines as native fs.symlink without type just works 
+// (see https://nodejs.org/api/fs.html#fssymlinktarget-path-type-callback). However on windows machines
+// we need to be more specific on whether or not we are attempting to symlink a file or directory.
+function getSymlinkArgs(src) {
```

Comment:
Can we put the fix into the existing `if (isNodeModulePath(src))` by any chance?

Reply:
Or have you seen the issue occur outside the `node_modules`?

Reply:
Well, I ran across this issue in the `node_modules` so that is the immediate thing that needs fixing. However, I don't think it's a good idea to limit the fix to `node_modules` only. Any time we call `fs.symlink` on windows machines we must specify the type otherwise we run the risk of this happening again for some other directory.

Reply:
Can you explain why we need the node_modules specific logic though instead of just `isDirectory`?